storage/src/test/java/org/apache/kafka/server/log/remote/storage/LocalTieredStorage.java

```diff
@@ -0,0 +1,564 @@
+
+                final InputStream inputStream = newInputStream(fileset.getFile(SEGMENT).toPath(), READ);
+                inputStream.skip(startPos);
+
+                // endPosition is ignored at this stage. A wrapper around the file input stream can implement
+                // the upper bound on the stream.
+
+                storageListeners.onStorageEvent(eventBuilder.withFileset(fileset).build());
+
+                return inputStream;
+
+            } catch (final Exception e) {
+                storageListeners.onStorageEvent(eventBuilder.withException(e).build());
+                throw e;
+            }
+        });
+    }
+
+    @Override
+    public InputStream fetchIndex(RemoteLogSegmentMetadata metadata, IndexType indexType) throws RemoteStorageException {
+        EventType eventType = getEventTypeForFetch(indexType);
+        RemoteLogSegmentFileset.RemoteLogSegmentFileType fileType = getLogSegmentFileType(indexType);
+        return wrap(() -> {
+            final LocalTieredStorageEvent.Builder eventBuilder = newEventBuilder(eventType, metadata);
+
+            try {
+                final RemoteLogSegmentFileset fileset = openFileset(storageDirectory, metadata.remoteLogSegmentId());
+
+                File file = fileset.getFile(fileType);
+                final InputStream inputStream = (fileType.isOptional() && !file.exists()) ?
+                        new ByteArrayInputStream(new byte[0]) : newInputStream(file.toPath(), READ);
+
+                storageListeners.onStorageEvent(eventBuilder.withFileset(fileset).build());
+
+                return inputStream;
+
+            } catch (final Exception e) {
+                storageListeners.onStorageEvent(eventBuilder.withException(e).build());
+                throw e;
+            }
+        });
+    }
+
+    @Override
+    public void deleteLogSegmentData(final RemoteLogSegmentMetadata metadata) throws RemoteStorageException {
+        wrap(() -> {
+            final LocalTieredStorageEvent.Builder eventBuilder = newEventBuilder(DELETE_SEGMENT, metadata);
+
+            if (deleteEnabled) {
+                try {
+                    final RemoteLogSegmentFileset fileset = openFileset(
+                            storageDirectory, metadata.remoteLogSegmentId());
+
+                    if (!fileset.delete()) {
+                        throw new RemoteStorageException("Failed to delete remote log segment with id:" +
+                                metadata.remoteLogSegmentId());
+                    }
+
+                    storageListeners.onStorageEvent(eventBuilder.withFileset(fileset).build());
+
+                } catch (final Exception e) {
+                    storageListeners.onStorageEvent(eventBuilder.withException(e).build());
+                    throw e;
+                }
+            }
+            return null;
+        });
+    }
+
+    // New API to be added in RemoteStorageManager
+    // @Override
```

Comment:
RemoteStorageManager does not have the `deletePartition` API. Once added, will remove this comment. This API is required to delete the partition in one shot instead of removing the segments one by one.

Reply:
I see. Do we have JIRA ticket tracking it?

Reply:
Filed: https://issues.apache.org/jira/browse/KAFKA-15166 to track it.